### ISSUE TITLE
Can't stream chart and keep the history of old items

### ISSUE THREAD

**Issue by artemiusgreat** [d3fc/d3fc#1593]
Hi, there is an example of how to create streaming chart 
[https://d3fc.io/examples/streaming-chart/](https://d3fc.io/examples/streaming-chart/)
Unfortunately, this example moves chart by the mean of removing old candles, so it would not be possible to see the history of previous candles that were removed. 
I created an example that doesn't remove old candles and chart is not moving anymore and is trying to scale down the whole chart to fit old and new candles in the same viewport... 
[https://codepen.io/artemiusgreat/pen/rNxPObo](https://codepen.io/artemiusgreat/pen/rNxPObo) 
**Instead of removing old candles, is it possible to enable panning and shift the viewport every time new item is added to the chart?**

**Comment by Serg65s** [comment]
data.slice(start, end) ?

**Comment by chrisprice** [comment]
The domain of an scale controls what is rendered within the viewable range. In your example xDomain is being set to xExtent so all of the data is visible. 

For pan/zoom see the examples e.g. https://d3fc.io/examples/chart-zoom/

**Comment by chrisprice** [comment]
Sorry didn't answer your actual question: yes it is possible using the techniques I mentioned in my comment.

**Comment by artemiusgreat** [comment]
@chrisprice Hi Chris, thank you for your answer. I tried to modify chart according to your example. Pan and zoom work when I initiate it with a mouse. Meanwhile, it's still unclear how to call zoom or pan programatically, from code? Your example shows how to pan canvas using `d3.event.transform`. What if I don't have an event? 

https://codepen.io/artemiusgreat/pen/jOqBJaa

**Comment by chrisprice** [comment]
In that case you just explicitly set the domain of the scales to the required values. In the example I previously linked, try running the following in the console `x.domain([-1, +1]); y.domain([-1, +1]); render()`.

**Comment by artemiusgreat** [comment]
@chrisprice Thank you very much for the tip. Only after printing `domain` values to the console I realized that this is an actual visible range of the chart.  

https://codepen.io/artemiusgreat/pen/jOqBJaa 

Still need to make mouse-events work together with auto-panning, but this is another story. Thanks.
